src/Init/Data/BitVec/Bitblast.lean

```diff
@@ -613,6 +613,12 @@ theorem neg_sle_zero (h : 0 < w) (x : BitVec w) :
   rw [sle_eq_slt_or_eq, neg_slt_zero h, sle_eq_slt_or_eq]
   simp [Bool.beq_eq_decide_eq (-x), Bool.beq_eq_decide_eq _ x, Eq.comm (a := x), Bool.or_assoc]
 
+theorem sle_eq_ule (x y : BitVec w) : x.sle y = (x.msb != y.msb ^^ x.ule y) := by
+  rw [sle_eq_not_slt, slt_eq_ult, ← Bool.xor_not, ← ule_eq_not_ult, bne_comm]
```

Comment:
```suggestion
theorem sle_eq_ule {x y : BitVec w} : x.sle y = (x.msb != y.msb ^^ x.ule y) := by
  rw [sle_eq_not_slt, slt_eq_ult, ← Bool.xor_not, ← ule_eq_not_ult, bne_comm]
```

Reply:
That would be inconsistent with `slt_eq_ult`, which was added 10 months ago. Would you like me to change both?

Reply:
You are an expert on the lean std API. Can you let me know which rules you follow, then we can apply those throughout. @kim-em suggested to use implicit arguments for such theorems, but I am happy to update the guidelines.

We are not super consistent, but I would be happy to review the full BitVec files to make sure we are consistent. As we now polish the API, this is probably the best moment to do this.

Reply:
Unfortunately we haven't been able to finalize an official policy for implicitness of arguments yet, but this is in the works. I'll go with implicits here for now.